backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/db/Database.java

```diff
@@ -100,17 +104,25 @@ public static String TO_DATE(@Nullable final Timestamp time, final boolean dayOn
 		}
 
 		final StringBuilder dateString = new StringBuilder("TO_TIMESTAMP('");
-		// YYYY-MM-DD HH24:MI:SS.mmmm JDBC Timestamp format
-		final String myDate = time.toString();
+		final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSSSSS")
```

Comment:
pls extract this as a class constant

Reply:
and IMHO I would introduce 2 constants,
* JDBC_DATETIME_FORMAT, i.e. DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSSSSS").withZone(ZoneOffset.UTC);
* JDBC_DATE_FORMAT, i.e. something similar but without time part.

so you won't need to do "dateString.append(myDate.substring(0, 10));"